tests/task_queue_test.go

```diff
@@ -164,6 +167,129 @@ func (s *TaskQueueSuite) testTaskQueueRateLimitName(nPartitions, nWorkers int, u
 	return "OldMatching_" + ret
 }
 
+func (s *TaskQueueSuite) TestTaskQueueApiLimitOverride() {
+	const buffer = 100 * time.Millisecond
+
+	s.Run("RateLimitTest_1.00", func() {
+		expectedGap := time.Second
+		s.runTestTaskQueueAPIRateLimitOverridesWorkerLimit(1.0, 5, expectedGap-buffer, expectedGap+buffer, "RateLimitTest_1.00")
+	})
+
+	s.Run("RateLimitTest_2.00", func() {
+		expectedGap := 500 * time.Millisecond
+		s.runTestTaskQueueAPIRateLimitOverridesWorkerLimit(2.0, 5, expectedGap-buffer, expectedGap+buffer, "RateLimitTest_2.00")
+	})
+}
+
+func (s *TaskQueueSuite) runTestTaskQueueAPIRateLimitOverridesWorkerLimit(apiRPS float32, taskCount int, minGap time.Duration, maxGap time.Duration, activityTaskQueue string) {
+	// Set the burst as 1 to make sure not more than 1 task get's acknowledged at a time.
+	// Helps observe the backlog drain more easily.
+	s.OverrideDynamicConfig(dynamicconfig.MatchingMinTaskThrottlingBurstSize, 1)
+	// Configure a single partition to ensure the full rate limit applies to one task queue
+	// without being divided across multiple partitions.
+	s.OverrideDynamicConfig(dynamicconfig.MatchingNumTaskqueueReadPartitions, 1)
+	s.OverrideDynamicConfig(dynamicconfig.MatchingNumTaskqueueWritePartitions, 1)
+	// Set a very low TTL for task queue info cache to ensure the rate limiter stats
+	// are refreshed frequently, avoiding stale data during the test.
+	s.OverrideDynamicConfig(dynamicconfig.TaskQueueInfoByBuildIdTTL, 1*time.Millisecond)
+
+	var (
+		mu       sync.Mutex
+		runTimes []time.Time
+	)
+
+	const (
+		workerRPS    = 50.0
+		drainTimeout = 30 * time.Second // Setting extremely high drain timeout to prevent flaking
+		activityName = "trackableActivity"
+	)
+
+	tv := testvars.New(s.T())
+	tv = tv.WithNamespaceName(namespace.Name(fmt.Sprintf("ns-rps-%.2f", apiRPS))) // To isolate state and avoid flaking
```

Comment:
I don't think this has any effect; since you need to use `s.Namespace().String()` anyway.